apps/aecore/test/aecore_sync_SUITE.erl

```diff
@@ -214,14 +243,45 @@ mine_and_compare(N1, Config) ->
               true = expect_block(Nx, NewTop)
       end, AllNodes -- [N1]).
 
-expect_same(Config) ->
-    Nodes = ?config(nodes, Config),
-    expect_same(Nodes, 5),
+expect_same(T0, Config) ->
+    Nodes = [N || {_, N} <- ?config(nodes, Config)],
+    [subscribe(N, chain_sync) || N <- Nodes],
+    AllEvents = lists:flatten(
+                  [events_since(N, chain_sync, T0) || N <- Nodes]),
+    ct:log("AllEvents = ~p", [AllEvents]),
+    Nodes1 =
+        lists:foldl(
+          fun(Msg, Acc) ->
+                  check_sync_event(Msg, Acc)
+          end, Nodes, AllEvents),
+    ct:log("Nodes1 = ~p", [Nodes1]),
+    collect_sync_events(Nodes1),
+    expect_same_top(Nodes, 5),
     expect_same_tx(Nodes).
 
-expect_same(Nodes, Tries) when Tries > 0 ->
+collect_sync_events([]) ->
+    done;
+collect_sync_events(Nodes) ->
+    receive
+        {gproc_ps, event, chain_sync, Msg} ->
```

Comment:
Type `gproc_ps, event` --> `gproc_ps_event` ? Ref https://github.com/uwiger/gproc/blob/0.6.1/doc/gproc_ps.md#publish3

This is compatible with local failure I got.

Reply:
mmm but how can this pass on CI?

Reply:
I've just pushed a fix for this. There are several timing issues in the sync suite, not least whether the test case has time to start checking for a sync event before the event has already been published. This is why the test proxy saves events and the test case collects and checks past events first. So tests may well pass in the case where events are published before the check happens, even though the following receive pattern is wrong.